src/components/Login.js

```diff
@@ -50,17 +51,33 @@ export const Login = props => {
   const classes = useStyles();
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
+
   useEffect(() => {
     return () => {
       setPassword("");
     };
   }, []);
+
+
   const handleSubmit = e => {
     e.preventDefault();
+    setError()
     const { history } = props;
-    props.auth(email, password, history);
-    setPassword("");
-    setEmail("");
+    setLoading(true);
+    props.auth(email, password).then(res => {
+      setPassword("");
+      setEmail("");
+      setLoading(false);
+      if (res === "success") {
+        history.push('/hacks')
```

Comment:
I prefer the action owning the business logic of where to go after logging in, rather than the component having it.

Reply:
I've made the changes, what is the reason for keeping the redirect in the action over the component?

Reply:
bit of preference, but also to keep the react component down the view logic only. domain logic (ie, the flow of the application) based on the actions outcome should be handled by the outside of the view.

Redux is trying to get us to remove as much as possible from the component itself, into its concept of actions and state. this simplifies the components, makes unit testing the results of actions easier 

view renders based on redux state
view says "do the thing"
action decides what doing the thing means, which may include updating redux state
view re-renders based on new state (if required)

Its the omni-direction data flow that makes redux so powerful and predictable.

Reply:
https://redux.js.org/basics/data-flow

Reply:
Can definitely appreciate separating the unit test from the view portion of React and test that in Redux!